tests/fixtures/testModel.ts

```diff
@@ -440,9 +440,31 @@ export interface GenericModel<T> {
   result: T;
   union?: T | string;
   nested?: GenericRequest<T>;
+  heritageCheck?: ThingContainerWithTitle<T>;
 }
 
 export interface GenericRequest<T> {
   name: string;
   value: T;
 }
+
+interface ThingContainerWithTitle<T> extends GenericContainer<number, number> {
+  // T is TestModel[] here
+  t: T;
+  title: string;
+}
+
+interface GenericContainer<T, U> {
+  id: string;
+  // T is number here
+  list: T[];
+  dangling: DanglingContext<T>;
+}
+
+interface DanglingContext<T> {
+  shouldBeString: U;
+}
+
+interface U {
```

Comment:
@WoH why is `U`  a non-generic interface on line 468 and then it `U` is used before as a generic type on line 457? Are they the same U (and if so, how is that possible?)

Reply:
That's part of what I'm trying to prove. They should not be.

Reply:
Okay that sounds good. I’m okay with the names, I just wanted t understand. Maybe there should be a comment in the code that explains this.

Reply:
My intention was to indicate that by naming the property "shouldBeString". However, I realize that's not sufficient. What would you add there? I feel like any attempt by me would not be as helpful because of language barrier and the fact that I'm very familiar with the code I wrote recently which would potentially lead to bias.

Reply:
Thank you for clarifying. :)

> What would you add there?

We could rename `U` to be `TSameNameDifferentValue` and then above the interface we could add a JSDoc comment with: "Although this has the same name as the generic type in GenericContainer, it is not the same since this type is concrete and the TSameNameDifferentValue within GenericContainer is a type parameter. We've created this scenario to clarify that tsoa should be able to tell there difference between the two."

Reply:
I did not use that exact wording, but I added comments, used `TSameNameDifferentValue` and hope to clarify for everyone what I'm doing.

Reply:
Cool yea, it was just a suggestion. I like what you wrote. Thank you for following up with that. :)

Btw, do you need published to npm? Or is it okay to sit in master until a new feature is released? (I can deploy this if you want people to start trying it)